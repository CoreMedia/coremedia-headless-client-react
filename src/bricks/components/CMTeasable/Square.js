import React from "react";

import Picture from "../CMPicture";
import Overlay from "./Overlay";
import OptionalLink from "../Common/OptionalLink";
import Strings from "../../../l18n";
import getColors from "../../../styles/colors";
import {TeaserBox} from "../../../styles";

const composeContent = (url, pictureLink: string, pictureTitle: string, pictureAlt: string, title?: string, text?: string, params?: any) => {
  const colors = getColors(params.color);
  const content = [];
  if (pictureLink) {
    content.push(
      <Picture key="picture"
               link={pictureLink}
               ratio="portrait_ratio1x1"
               title={pictureTitle}
               alt={pictureAlt}
               colors={colors}
      />
    );
  }
  if (title || text || params.ctaShow) {
    content.push(
      <Overlay key="overlay"
               width={100}
               height="45%"
               bottom="0%"
               title={title}
               text={text}
               colors={colors}
               ctaShow={params.ctaShow && !!url}
               ctaText={params.ctaText || Strings.teaser_cta_text}
      />
    );
  }
  return content;
};

type Props = {
  url?: string,
  pictureLink?: string,
  pictureTitle?: string,
  pictureAlt?: string,
  title?: string,
  text?: string,
  params?: any
};

const SquareTeaser = ({url, pictureLink, pictureTitle, pictureAlt, title, text, params = {},}: Props) => {
  return (
    <OptionalLink url={url}>
      <TeaserBox>
        {composeContent(url, pictureLink, pictureTitle, pictureAlt, title, text, params)}
      </TeaserBox>
    </OptionalLink>
  );
};

export default SquareTeaser;
