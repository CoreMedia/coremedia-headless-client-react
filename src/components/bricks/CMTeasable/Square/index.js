// @flow
import * as React from 'react';
import PropTypes from 'prop-types';

import Picture from '../../CMPicture';
import Overlay from '../Overlay';
import { TeaserBox } from '../../../basic/Box';
import { OptionalLink } from '../../../basic/Link';
import Strings from '../../../../l18n';
import type { Teaser } from '../../../../types';

const composeContent = ({
  url,
  pictureLink,
  pictureTitle,
  pictureAlt,
  title,
  text,
  params,
}: Teaser): React.Node => {
  const color = params && params.color;
  const content = [];
  if (pictureLink) {
    content.push(
      <Picture
        key="picture"
        link={pictureLink}
        ratio="portrait_ratio1x1"
        title={pictureTitle}
        alt={pictureAlt}
        color={color}
      />
    );
  }
  if (title || text || (params && params.ctaShow)) {
    content.push(
      <Overlay
        key="overlay"
        width={100}
        height="45%"
        bottom="0%"
        title={title}
        text={text}
        color={color}
        ctaShow={!!params && !!params.ctaShow && !!url}
        ctaText={(params && params.ctaText) || Strings.teaser_cta_text}
      />
    );
  }
  return content;
};

const SquareTeaser = ({
  url,
  pictureLink,
  pictureTitle,
  pictureAlt,
  title,
  text,
  params = {},
}: Teaser) => (
  <OptionalLink url={url}>
    <TeaserBox>
      {composeContent({ url, pictureLink, pictureTitle, pictureAlt, title, text, params })}
    </TeaserBox>
  </OptionalLink>
);

SquareTeaser.propTypes = {
  url: PropTypes.string,
  pictureLink: PropTypes.string,
  pictureTitle: PropTypes.string,
  pictureAlt: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string,
  params: PropTypes.shape({
    color: PropTypes.string,
    ctaShow: PropTypes.bool,
    ctaText: PropTypes.string,
    url: PropTypes.string,
  }),
};

export default SquareTeaser;
