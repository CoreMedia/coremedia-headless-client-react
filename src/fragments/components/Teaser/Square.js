import React from "react";

import {Square as TeaserBrick} from "../../../bricks/components/CMTeasable";

const SquareTeaser = {
  type: 'teasables',
  view: null,
  brick(data, params) {
    const picture = data.picture || {};
    return (
      <TeaserBrick url={params.url ? params.url : data.teaserTarget.link}
                   pictureLink={picture.link}
                   pictureTitle={picture.title}
                   pictureAlt={picture.alt}
                   title={data.teaserTitle}
                   text={data.teaserText}
                   params={params}
      />
    );
  }
};

export default SquareTeaser;
