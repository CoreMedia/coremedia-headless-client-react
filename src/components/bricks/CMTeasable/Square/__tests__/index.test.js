// @flow
import React from 'react';

import Square from '..';

describe('SquareTeaser Component', () => {
  it('should render SquareTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link and Picture', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link and Overlay', () => {
    const wrapper = shallow(
      <Square params={{ color: 'blue', ctaShow: true, url: '/caas.html' }} url="/caas.html" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Link', () => {
    const wrapper = shallow(
      <Square params={{ color: 'blue', url: '/caas.html' }} url="/caas.html" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Overlay', () => {
    const wrapper = shallow(
      <Square params={{ color: 'blue', ctaShow: true, url: '/caas.html' }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Picture', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Picture (without params)', () => {
    const wrapper = shallow(
      <Square pictureAlt="Globe" pictureLink="coremedia:///image/2656/data" pictureTitle="Globe" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SquareTeaser with Picture and Overlay', () => {
    const wrapper = shallow(
      <Square
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
