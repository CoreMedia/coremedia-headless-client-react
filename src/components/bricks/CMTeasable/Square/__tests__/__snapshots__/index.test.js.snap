// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareTeaser Component should render SquareTeaser with Link 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox />
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link and Overlay 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={true}
      ctaText="More"
      height="45%"
      key="overlay"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link and Picture 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link, Picture and Overlay 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={true}
      ctaText="More"
      height="45%"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link, Picture and Overlay 2`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={false}
      ctaText="More"
      height="45%"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link, Picture and Overlay 3`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={false}
      ctaText="More"
      height="45%"
      key="overlay"
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link, Picture and Overlay 4`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={false}
      ctaText="More"
      height="45%"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link, Picture and Overlay 5`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={true}
      ctaText="More"
      height="45%"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link, Picture and Overlay 6`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={true}
      ctaText="More"
      height="45%"
      key="overlay"
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Link, Picture and Overlay 7`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={true}
      ctaText="More"
      height="45%"
      key="overlay"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Overlay 1`] = `
<OptionalLink>
  <TeaserBox>
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={false}
      ctaText="More"
      height="45%"
      key="overlay"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Picture (without params) 1`] = `
<OptionalLink>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Picture 1`] = `
<OptionalLink>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`SquareTeaser Component should render SquareTeaser with Picture and Overlay 1`] = `
<OptionalLink>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="portrait_ratio1x1"
      title="Globe"
    />
    <Overlay
      bottom="0%"
      color="blue"
      ctaShow={false}
      ctaText="More"
      height="45%"
      key="overlay"
      width={100}
    />
  </TeaserBox>
</OptionalLink>
`;
